lib/loading.dart

```diff
@@ -46,12 +47,11 @@ class LoadingOverlay {
     }
   }
 
-  OverlayEntry _createdLoadingEntry(BuildContext context) {
+  OverlayEntry _createdLoadingEntry(BuildContext context, bool blockGestures) {
     return OverlayEntry(
-      builder: (BuildContext context) => Stack(
-        children: <Widget>[
-          if (loadingUi != null) loadingUi! else Loading(),
-        ],
+      builder: (BuildContext context) => AbsorbPointer(
+        absorbing: blockGestures,
```

Comment:
And when I mentioned adding a option, I meant adding it globally to the constructor

```dart
const GlobalDialog({Key? key, required this.child, this.loadingOverlay})
      : super(key: key);
```